csm_web/frontend/src/components/Home.tsx

```diff
@@ -2,20 +2,47 @@ import React from "react";
 import { Link } from "react-router-dom";
 import { groupBy } from "lodash";
 import { useProfiles } from "../utils/queries/base";
+import { useCourses } from "../utils/queries/courses";
 import { ROLES } from "./section/Section";
-import { Profile } from "../utils/types";
+import { Profile, Course } from "../utils/types";
 import LoadingSpinner from "./LoadingSpinner";
 
 const Home = () => {
   const { data: profiles, isSuccess: profilesLoaded, isError: profilesLoadError } = useProfiles();
+  const { data: courses, isSuccess: coursesLoaded, isError: coursesLoadError } = useCourses();
 
   let content = null;
-  if (profilesLoaded) {
+  if (profilesLoaded && coursesLoaded) {
     // loaded, no error
+    const coursesById: Map<number, Course> = new Map();
+    for (const courseObj of courses) {
+      coursesById.set(courseObj.id, courseObj);
+    }
+
     content = (
       <div className="course-cards-container">
-        {Object.entries(groupBy(profiles, (profile: Profile) => [profile.course, profile.role])).map(
-          ([course, courseProfiles]) => {
+        {Object.entries(groupBy(profiles, (profile: Profile) => [profile.course, profile.role]))
+          .sort(([courseNameA, courseProfilesA], [courseNameB, courseProfilesB]) => {
+            const compare = (a: string, b: string) => {
+              if (a < b) {
+                return -1;
+              } else if (a > b) {
+                return 1;
+              } else {
+                return 0;
+              }
+            };
```

Comment:
lol this is funny. uhhh i don't know if this works https://stackoverflow.com/questions/51165/how-to-sort-strings-in-javascript 
```
list.sort(function (a, b) {
    return ('' + a.attr).localeCompare(b.attr);
})
``` 
but if it does, it looks less cursed.